### ISSUE TITLE
Feature requests: better behavior for debugging plugins

### ISSUE THREAD

**Issue by jni** [napari/napari#2841]
## 🚀 Feature

When plugins have errors, it is currently a frustrating experience to debug them in napari:

- [ ] To find plugin errors, I have to go to "Plugins > Plugin Errors > Select plugin...". That last click in particular is totally unnecessary: if there are one or more items in that dropdown menu, it should load the first item automatically.
- [ ] Errors are not surfaced in the notification manager or the console. Reasonable for users, but napari should have a debug mode in which plugin errors are not captured.
- [ ] Errored plugins don't appear in the dock widget list. This is confusing. I suggest that they instead should still appear, but grayed out with a :warning: symbol. Clicking on them would take you to the error dialog.
- [ ] Plugins are disabled after the first error. So, if I go and fix my plugin, and run napari again, I have to go to the plugins menu and reactivate it. Although this is reasonable from a user's perspective, it's a frustrating experience for plugin developers.

**Comment by tlambert03** [comment]
I agree with all these things.  for what it's worth, I use `napari --plugin-info -v` a _lot_ when debugging plugins and it satisfies some of these things.

**Comment by sofroniewn** [comment]
I just read through image.sc link above - sound like this would also fix some bugs that folks ran into over there. Might this be a good thing for @ppwadhwa to work on @tlambert03 or do you have other ideas for a fix?

**Comment by sofroniewn** [comment]
@nclack could be a good one for you to look into
